### ISSUE TITLE
Change option from cancel to no for delete directory request when removing a project in VS Code

### ISSUE THREAD

**Issue by beccabau** [eclipse/codewind#2105]
**Codewind version:** 0.8.0
**OS:** macos catalina

**IDE version:** vs code

**Description of the enhancement:**

![image](https://user-images.githubusercontent.com/13971834/74158591-5a6fe580-4be8-11ea-92b4-5ecff239d49e.png)

When I remove a project from local codewind in vs code, I also have the option to delete the directory when removing a project, but sometimes I might not want to. I found that clicking the "Cancel" button works the same as saying "No, don't delete the directory."

**Proposed solution:** 

Can we change the button to say "no" instead of "cancel?" Cancel makes is sound like I want to cancel the entire action (even though it proceeds to remove the project, but keep the directory)

**Comment by tetchel** [comment]
I agree this is confusing, but there always has to be a Cancel button, there's no way to turn it off.
I guess I could do `Delete Directory | Don't Delete | Cancel` and then cancel the entire deletion if they click Cancel at that step. But, I already asked them to confirm the delete in the previous dialog 🤔

**Comment by beccabau** [comment]
For me, the first prompt is asking to confirm project removal, which is a different from deleting the directory (right?)

![image](https://user-images.githubusercontent.com/13971834/74159795-58a72180-4bea-11ea-828d-43816e1774d1.png)

**Comment by tetchel** [comment]
yes, that's why i'm saying it wouldn't make sense to have cancelling the second step go back and also cancel the first step

**Comment by tetchel** [comment]
Basically, keeping in mind that a Cancel button always has to be shown, how would you design this?

**Comment by beccabau** [comment]
sorry, maybe I'm miscommunicating, so re: 
```
I guess I could do `Delete Directory | Don't Delete | Cancel` and then cancel the entire deletion if they click Cancel at that step. 
```
^ This is the second dialog, right? 

So the first prompt is `remove project | cancel`
Can the second prompt have 3 actions? `Delete Directory | Don't Delete | Cancel` (where cancel will revert the entire process if clicked)? That's how I understood the cancel button when I first saw it.

**Comment by tetchel** [comment]
yea, i think that is a good change

**Comment by tetchel** [comment]
This is how VS Code organizes the dialog if there are 2 buttons (not counting built-in Cancel). The blue button is the default. What should be default, Delete or Don't Delete?

**Comment by thisguy-1** [comment]
this is tough. In my opinion, I'd change the message to something like (@sishida please give opinion):
First dialogue should say something along the lines of:
"are you sure you want to remove <project> from <connection> from the codewind extension? By selecting remove you will no longer see it in your codewind view"
Second dialogue:
"You have successfully removed <project> from <connection> and will no longer see it in your codewind view. Would you also like to remove its project directory <directory>?"
Maybe also adding  "Select Cancel to keep the directory in your system".

**Comment by tetchel** [comment]
btw we can always make it one dialog.

Remove | Remove and delete files | Cancel

**Comment by thisguy-1** [comment]
I'd rather keep with two. It feels more familiar.

**Comment by sishida** [comment]
What about something like...
- First dialog message:`Are you sure you want to remove <project> from <connection> from the Codewind extension? If you click Remove, the project directory is removed from your Codewind view.`
- First dialog buttons: `Remove directory` `Cancel` 

- Second dialog message: `You successfully removed <project> from <connection> and will no longer see <project> in your Codewind view. Do you also want to remove the <directory> project directory? Click Cancel to keep the directory on your computer.`
- Second dialog buttons: `Remove directory` `Cancel`

**Comment by thisguy-1** [comment]
> What about something like...
> 
> * First dialog message:`Are you sure you want to remove <project> from <connection> from the Codewind extension? If you click Remove, the project directory is removed from your Codewind view.`
> * First dialog buttons: `Remove directory` `Cancel`
> * Second dialog message: `You successfully removed <project> from <connection> and will no longer see <project> in your Codewind view. Do you also want to remove the <directory> project directory? Click Cancel to keep the directory on your computer.`
> * Second dialog buttons: `Remove directory` `Cancel`

i think the first dialogue should not refer to the item being removed as "project directory". it should just be project.

**Comment by thisguy-1** [comment]
and the first dialogue button should be Remove Project and the second should be Delete Directory.

**Comment by tetchel** [comment]
It's okay, the messages are far too wordy for my liking - but mostly it doesn't address the original concern, which is that "cancel" in the second dialog is expected to cancel the entire removal process. 
I understand the message specifies what Cancel does / doesn't do, but I think most won't read that far.

**Comment by thisguy-1** [comment]
I think the first phrase of the second dialogue fixes that. It confirms that the previous action is done. If they run into the problem of not knowing what cancel does, the last phrase deals with that. The original concern was that a user would have the doubt of not knowing what cancel does, whether it cancels all the work done before or just cancels the deletion of the directory. I think this is our best option rather than a very unconventional set of buttons that might confuse and cause mistakes.

**Comment by thisguy-1** [comment]
--given that we can't change the wording for the "cancel" button.

**Comment by sishida** [comment]
Because we're stuck with using the "Cancel" button, I liked explicitly stating what it was doing.

However, if we're thinking users will ignore it, what if we gave the statement its own dialog--is that too obnoxious? 😆Something like...`If you click Cancel instead of proceeding to remove both the project and the project directory, the removal for both is cancelled.`

...I realize that doesn't help with the wordiness, however.

**Comment by tetchel** [comment]
ok, then let's work with this:
```
Are you sure you want to remove <project> from <connection>? If you click Remove, the project  is removed from your Codewind view.
```
> the project is removed from your Codewind view.
I don't really like this because it's more like we're removing the project from the connection (as already stated). for example, the project may still be in the codewind view under a different connection. I would rather leave out the second sentence.

The second one is fine if we simplify it:
```
You successfully removed <project> from <connection>. Do you also want to delete <directory path> from your disk? Click Cancel to keep the directory.
```

**Comment by tetchel** [comment]
i guess one problem is that "you successfully removed" is misleading because we don't start the remove until after both dialogs are closed. so maybe `<project> will be removed from <connection>.` instead

**Comment by j-c-berger** [comment]
@tetchel, that suggestion sounds good to me. I'm happy to do a quick review of the change before we incorporate it.

**Comment by tetchel** [comment]
![image](https://user-images.githubusercontent.com/8888438/74461349-6532b080-4e5c-11ea-8ede-4f4006fd327a.png)

**Comment by j-c-berger** [comment]
@tetchel, looks good! See some quick comments I made on your commit, `Update project removal dialog message` Thanks!

**Comment by tetchel** [comment]
Those are old messages previously approved by design/ID, I just changed 'containers' to 'images' because it is more technically correct
